packages/module/src/WarningModal/WarningModal.tsx

```diff
@@ -1,13 +1,19 @@
-import React from 'react';
-import { Button, ModalProps, Modal, ModalVariant, ButtonVariant } from '@patternfly/react-core';
+import React, { useState } from 'react';
+import { Button, ModalProps, Modal, ModalVariant, ButtonVariant, Checkbox } from '@patternfly/react-core';
 
 export interface WarningModalProps extends Omit<ModalProps, 'ref'> {
   /** Callback for the confirm action button. */
   onConfirm?: () => void;
   /** Custom label for the confirm action button */
-  confirmButtonLabel? : string;
+  confirmButtonLabel?: string;
   /** Custom label for the cancel action button */
-  cancelButtonLabel? : string;
+  cancelButtonLabel?: string;
+  /** Whether Modal requires a checkbox before confirming */
+  withCheckbox?: boolean;
+  /** Custom message after confirmation */
+  confirmCheckMessage?: string;
```

Comment:
I would rename it to checkboxLabel since checkbox uses label as well
```suggestion
  /** Custom checkbox label */
  checkboxLabel?: string;
```